etcdctl/diagnosis/engine/diagnosis.go

```diff
@@ -0,0 +1,47 @@
+package engine
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+
+	"go.etcd.io/etcd/etcdctl/v3/diagnosis/engine/intf"
+)
+
+const (
+	reportFileName = "etcd_diagnosis_report.json"
```

Comment:
I suggest that we add a flag `--output` (short `-o`) so that users can set the output file name.

Reply:
Done. Added -o/--output. When provided, the JSON report is written to the given file; otherwise it’s printed to stdout.